apps/nextjs/src/app/onboarding/onboarding.tsx

```diff
@@ -1,128 +1,46 @@
 "use client";
 
-import { useState } from "react";
+import { useEffect, useRef } from "react";
 
+import { useUser } from "@clerk/nextjs";
 import logger from "@oakai/logger/browser";
-import {
-  OakBox,
-  OakFlex,
-  OakHeading,
-  OakLink,
-  OakP,
-  OakPrimaryButton,
-  OakSpan,
-} from "@oaknational/oak-components";
-import Link from "next/link";
+import { useReloadSession } from "hooks/useReloadSession";
 
-import Button from "@/components/Button";
-import CheckBox from "@/components/CheckBox";
-import SignUpSignInLayout from "@/components/SignUpSignInLayout";
-import TermsContent from "@/components/TermsContent";
+import { AcceptTermsForm } from "@/components/Onboarding/AcceptTermsForm";
+import { LegacyUpgradeNotice } from "@/components/Onboarding/LegacyUpgradeNotice";
 import { trpc } from "@/utils/trpc";
 
 export const OnBoarding = () => {
-  const [dropDownOpen, setDropDownOpen] = useState(true);
-  const [termsAcceptedLocal, setTermsAcceptedLocal] = useState(false);
-  const [privacyAcceptedLocal, setPrivacyAcceptedLocal] = useState(false);
-  const acceptTerms = trpc.auth.acceptTerms.useMutation();
-
-  const handleAcceptTermsOfUse = async () => {
-    try {
-      const response = await acceptTerms.mutateAsync({
-        termsOfUse: new Date(),
-        privacyPolicy: privacyAcceptedLocal ? new Date() : false,
-      });
-
-      if (!response?.acceptedTermsOfUse) {
-        throw new Error("Could not accept terms of use");
+  const { user } = useUser();
+  const reloadSession = useReloadSession();
+  const setDemoStatus = trpc.auth.setDemoStatus.useMutation();
+
+  const userHasAlreadyAcceptedTerms =
+    user?.publicMetadata?.["labs"]?.["isOnboarded"];
```

Comment:
is this a TS complaint that the keys might not exist?

Reply:
Yeah. It's a slightly hokey way to wait for `user` to have loaded and then dig into those fields if they don't exist. `publicMetadata` is `unknown` here, so we have to access them as fields rather than properties